spring-cloud-gcp-test/src/main/java/org/springframework/cloud/gcp/test/PubSubEmulator.java

```diff
@@ -87,7 +86,7 @@ public PubSubEmulator() {
 	@Override
 	protected void before() throws IOException, InterruptedException {
 
-		assumeTrue("PubSubEmulator rule disabled. Please enable with -Dit.pubsub-emulator.", this.enableTests);
+		Assume.assumeTrue("PubSubEmulator rule disabled. Please enable with -Dit.pubsub-emulator.", this.enableTests);
```

Comment:
I don't think that enabling it with `-Dit.pubsub-emulator` makes sense when you're offering it as a generic test utility. That flag was specific for our project.

Reply:
It's an interesting question then to test whether the rule initialization or a test's `@BeforeClass` will run first. We would not want to do emulator startup if the tests themselves are disabled.

If there is possibility of the rule running first, perhaps we can just document this flag well?

Reply:
We can also try not to implement JUnit test rule.
That's what we ended up doing for Spring Cloud Function [here](https://github.com/spring-cloud/spring-cloud-function/blob/master/spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/test/java/org/springframework/cloud/function/adapter/gcp/integration/LocalServerTestSupport.java).